Adds new ability to add declarative config with a string feature (#3019)
Kong/docs.konghq.com: app/gateway-oss/2.5.x/db-less-and-declarative-config.md

## Loading The Declarative Configuration File

There are two ways to load a declarative configuration file into Kong: using
`kong.conf` or the Admin API.

To load a declarative configuration file at Kong start-up, use the
`declarative_config` directive in `kong.conf` (or, as usual to all `kong.conf`
entries, the equivalent `KONG_DECLARATIVE_CONFIG` environment variable).

```

> export KONG_DATABASE=off /
> export KONG_DECLARATIVE_CONFIG=kong.yml /

Reviewer: ```suggestion
export KONG_DATABASE=off \
export KONG_DECLARATIVE_CONFIG=kong.yml \
```

kong start -c kong.conf
```

You can also load a declarative configuration file into a running

> Kong node by its Admin API, using the `/config` endpoint. The

Reviewer 2: ```suggestion
Kong node with the Admin API, using the `/config` endpoint. The
```
this is the form I've been using elsewhere. lmk if you'd like to chat about an alternative.

following example loads `kong.yml` using HTTPie:

```
$ http :8001/config config=@kong.yml
```

{:.important}
> The `/config` endpoint replaces the entire set of entities in memory
with the ones specified in the given file.

Alternatively, you can start Kong in DB-less mode with a minimum

> declarative configuration in a string using the `KONG_DECLARATIVE_CONFIG`

Reviewer: ```suggestion
declarative configuration in a string using the `KONG_DECLARATIVE_CONFIG_STRING`
```

> Alternatively, you can start Kong in DB-less mode with a minimum
> declarative configuration in a string using the `KONG_DECLARATIVE_CONFIG_STRING`
> environment variable. 

Reviewer 2: This feels like it's heading into run-on territory (hasn't quite landed yet, but here's my effort). Making it a bit more casual too, in the spirit of conversational tone:

```suggestion
Or to start in DB-less mode, provide a minimum declarative configuration with a string you pass with the `KONG_DECLARATIVE_CONFIG_STRING` environment variable. 
```

```
docker run \
-e KONG_DATABASE=off \
-p 8000:8000 \
-v $HOME/git/kong/kong/db/declarative/init.lua:/usr/local/share/lua/5.1/kong/db/declarative/init.lua \
-e KONG_DECLARATIVE_CONFIG='{"_format_version":"1.1","services":[{"host":"mockbin.com","port":443,"protocol":"https","routes":[{"paths":["/"]}]}],"plugins":[{"name":"rate-limiting","config":{"policy":"local","limit_by":"ip","minute":3}}]}' \

> kong:latest

Reviewer: export KONG_DATABASE=off \
export KONG_DECLARATIVE_CONFIG_STRING='{"_format_version":"1.1","services":[{"host":"mockbin.com","port":443,"protocol":"https","routes":[{"paths":["/"]}]}],"plugins":[{"name":"rate-limiting","config":{"policy":"local","limit_by":"ip","minute":3}}]}' \
kong start

Author: will update
